### ISSUE TITLE
No progress updates from flatpak_installation_install_bundle()

### ISSUE THREAD

**Issue by hughsie** [flatpak/flatpak#404]
The libflatpak API call flatpak_installation_install_bundle() takes a FlatpakProgressCallback as an input, but never actually calls it. This means when you install a large .flatpak application (for instance, LibreOffice) you don't get any indication of progress. The progress bar goes from 0% and them jumps to 100% after about 40 seconds. This isn't awesome.

Looking at the code, the progress and progress_data variables look completely unused, so I guess they need to be passed to flatpak_dir_install_bundle() and handled in the system helper.

Thanks!

**Comment by alexlarsson** [comment]
The reason they are not passed on is that in the end we're calling into ostree to do a local "apply static delta" operation, and there is no progress reporting in ostree for this.

**Comment by hughsie** [comment]
But you'd agree that reporting no progress for 40 seconds is a bug, right? Surely ostree knows how many parts there are to deploy in the delta?

**Comment by alexlarsson** [comment]
Yes, ostree knows how large the files are etc. I guess the problem is that ostree doesn't consider "local file operations" to be "slow", because only things that do network i/o report progress.

@cgwalters any chance we could get progress reporting also for local operations like checkouts and applying static deltas?
